packages/api/src/perma-cache/get.js

```diff
@@ -19,10 +20,39 @@ export async function permaCacheGet(request, env) {
   const normalizedUrl = getNormalizedUrl(sourceUrl, env)
   const r2Key = normalizedUrl.toString()
 
-  const r2Object = await env.SUPERHOT.get(r2Key)
+  // Get range header
+  const range = parseRange(request.headers.get('range'))
+  const r2Object = await env.SUPERHOT.get(r2Key, {
+    range,
+  })
   if (r2Object) {
-    return new Response(r2Object.body)
+    return new Response(r2Object.body, {
+      status: range ? 206 : 200,
+    })
   }
 
   throw new UrlNotFoundError()
 }
+
+/**
+ *
+ * @param {string|null} encoded
+ * @returns {R2Range}
+ */
+function parseRange(encoded) {
+  if (encoded === null) {
+    return
+  }
+
+  const parts = encoded.split('bytes=')[1]?.split('-') ?? []
+  if (parts.length !== 2) {
+    throw new InvalidRangeError(
+      'Not supported to skip specifying the beginning/ending byte at this time'
```

Comment:
Can maybe use this: https://github.com/nftstorage/dagcargo-bucket/blob/main/src/r2.js